src/main/java/com/eerussianguy/blazemap/engine/async/DebouncingDelay.java

```diff
@@ -5,6 +5,7 @@
 public class DebouncingDelay {
     private static final int MAX_POOL = 1000;
     private static final ArrayList<DebouncingDelay> POOL = new ArrayList<>();
+    private static final int FUZZ_FACTOR = 5;
```

Comment:
I left in the changes I'd already made to this file, but am happy to leave them out. The intention is to always offset the exact tick the updates happen to minimise the load on any one particular tick, and thereby reduce the chance of hitching.